nixos/modules/services/networking/bee-clef.nix

```diff
@@ -0,0 +1,76 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.bee-clef;
+in {
+  meta = {
+    maintainers = with maintainers; [ attila-lendvai ];
+  };
+
+  ### interface
+
+  options = {
+    services.bee-clef = {
+      enable = mkEnableOption "clef external signer instance for Ethereum Swarm Bee";
+
+      data-dir = mkOption {
+        type = types.nullOr types.str;
+        default = "/var/lib/bee-clef";
+        description = "Data dir for bee-clef.";
+      };
+
+      password-file = mkOption {
+        type = types.nullOr types.str;
+        default = "/var/lib/bee-clef/password";
+        description = "Password file for bee-clef.";
+      };
+    };
+  };
+
+  ### implementation
+
+  config = mkIf cfg.enable {
+    systemd.packages = [ pkgs.bee-clef ];
+    # if we ever want to have rules.js under /etc/bee-clef/
+    # environment.etc."bee-clef/rules.js".source = ${pkgs.bee-clef}/rules.js
+
+    services.bee-clef = {
+      data-dir      = "/var/lib/bee-clef";
+      password-file = "/var/lib/bee-clef/password";
+    };
+
+    systemd.tmpfiles.rules = [
+        "d '${cfg.data-dir}' 0750 bee-clef bee-clef"
+        "d '${cfg.data-dir}/keystore' 0700 bee-clef bee-clef"
+      ];
+
+    systemd.services.bee-clef = {
+      wantedBy = [ "bee.service" ];
+      serviceConfig = {
+        ExecStart = [
+          "${pkgs.bee-clef}/clef-service start"
+        ];
+        ExecStop = [
+          "${pkgs.bee-clef}/clef-service stop"
+        ];
+        # NOTE this would create the dir with the wrong permissions
+        #StateDirectory = "${cfg.data-dir}";
+      };
+
+      # NOTE for now nothing is installed into /etc/bee-clef/. these files are used as read-only from the packages.
+      preStart = with cfg; ''
+        ${pkgs.bee-clef}/ensure-clef-account "${data-dir}" "${pkgs.bee-clef}"
+      '';
+    };
+
+    users.users.bee-clef = {
```

Comment:
The user name should be configurable and a custom value should prevent an user from being created automatically.
Is `home = cfg.data-dir` the only workable solutions here?

Reply:
A configurable user is not a requirement. The module author can decide what to do here.

Reply:
I think it is a good practice for the (theoretical) case that there is a name clash of sorts.

Reply:
it's hardwired at various place in the upstream files, so i'll leave user/group config commented out for now

Reply:
This can be overridden, but adding this option or not is the module authors choice. So feel free to leave out if you wish.